docs/java-rest/high-level/java-builders.asciidoc

```diff
@@ -0,0 +1,26 @@
+[[java-rest-high-java-builders]]
+== Using Java Builders
+
+The Java High Level REST Client depends on the Elasticsearch core project which provides
+two types of Java `Builders` objects:
+
+Query Builders::
+
+The query builders are used to create `QueryBuilder` objects which can later be set as the
```

Comment:
I think this is a bit confusing, at least to me. Query builders "are" the the implementations of the "QueryBuilder" interface, they don't create them but rather later create the Lucene queries on the shards. We don't need to mention that here, maybe better say that the query builders are used to describe the full query and implement the "QueryBuilder" interface, that is used in the SearchSourceBuilder? Or something else along those lines.